codegen/src/main/ops/org/nd4j/codegen/ops/SDMath.kt

```diff
@@ -3,2055 +3,1344 @@
         }
     }
 
     Op("amean") {
-        javaPackage = namespaceJavaPackage
-        Input(NUMERIC, "in") { description = "" }
-        Input(NUMERIC, "dimensions") { description = "" }
-
-        Output(NUMERIC, "output"){ description = "" }
-
+        javaPackage = "org.nd4j.linalg.api.ops.impl.reduce.floating"
+        Input(NUMERIC, "in") { description = "Input variable" }
+        Arg(INT, "dimensions"){ count = AtLeast(1); description = "Dimensions to reduce over. If dimensions are not specified, full array reduction is performed" }
+        Output(NUMERIC, "output"){ description = "Reduced array of rank (input rank - num dimensions)" }
         Doc(Language.ANY, DocScope.ALL){
             """
- Absolute mean array reduction operation, optionally along specified dimensions: out = mean(abs(x))
-
- @param name       Name of the output variable
- @param in         Input variable
- @param dimensions Dimensions to reduce over. If dimensions are not specified, full array reduction is performed
- @return Reduced array of rank (input rank - num dimensions)
-     
-""".trimIndent()
+                Absolute mean array reduction operation, optionally along specified dimensions: out = mean(abs(x))
+            """.trimIndent()
         }
     }
 
+    // TODO: There are 2 implementations of amax in org.nd4j.linalg.api.ops.impl
     Op("amin") {
-        javaPackage = namespaceJavaPackage
-        Input(NUMERIC, "in") { description = "" }
-        Input(NUMERIC, "dimensions") { description = "" }
-
-        Output(NUMERIC, "output"){ description = "" }
-
+        javaPackage = "org.nd4j.linalg.api.ops.impl.transforms.same"  // ""org.nd4j.linalg.api.ops.impl.reduce.same"
+        Input(NUMERIC, "in") { description = "Input variable" }
+        Arg(INT, "dimensions"){ count = AtLeast(1); description = "Dimensions to reduce over. If dimensions are not specified, full array reduction is performed" }
+        Output(NUMERIC, "output"){ description = "Reduced array of rank (input rank - num dimensions)" }
         Doc(Language.ANY, DocScope.ALL){
             """
- Absolute min array reduction operation, optionally along specified dimensions: out = min(abs(x))
-
- @param name       Name of the output variable
- @param in         Input variable
- @param dimensions Dimensions to reduce over. If dimensions are not specified, full array reduction is performed
- @return Reduced array of rank (input rank - num dimensions)
-     
-""".trimIndent()
+                Absolute min array reduction operation, optionally along specified dimensions: out = min(abs(x))
+            """.trimIndent()
         }
     }
 
     Op("and") {
-        javaPackage = namespaceJavaPackage
-        Input(NUMERIC, "x") { description = "" }
-        Input(NUMERIC, "y") { description = "" }
-
-        Output(NUMERIC, "output"){ description = "" }
-
+        javaPackage = "org.nd4j.linalg.indexing.conditions"
+        Input(NUMERIC, "x") { description = "Input 1" }
```

Comment:
If I see that correctly, the inputs here should have BOOL types. 

Also I think the correct java package should be `org.nd4j.linalg.api.ops.impl.transforms.pairwise.bool`